### PR TITLE
Cleared inputs file with/without LLG

### PR DESCRIPTION
**Pull request by jackieyao0114**
1. Added functionality of excluding `B` field initialization;
2. Cleared out all inputs files by 1. renaming with the key work LLG for `USE_LLG=TRUE`; 2. running all the files to make sure they predict results as expected. 
3. Note: `inputs_3d_LLG_tfmu` does not work currently, debugging in progress.

### CODE REVIEW

File: Examples/Tests/Macroscopic_Maxwell/inputs_3d_PR10 [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+warpx.Ez_excitation_grid_function(x,y,z,t) = "0.0"
+warpx.Ex_excitation_flag_function(x,y,z) = "flag_none"
+warpx.Ey_excitation_flag_function(x,y,z) = "flag_hs * (z>=h) * (z<h+6.0e-4) + flag_none * (z<h) * (z>=h+6.0e-4)"
+warpx.Ez_excitation_flag_function(x,y,z) = "flag_none"
+
+warpx.B_ext_grid_init_style = parse_B_ext_grid_function
+warpx.Bx_external_grid_function(x,y,z)= "0.0"
+warpx.By_external_grid_function(x,y,z)= 0.
+warpx.Bz_external_grid_function(x,y,z) = 0.
+
```
Comment (ajnonaka) [inline Examples/Tests/Macroscopic_Maxwell/inputs_3d_PR10:72]: For these USE_LLG=FALSE inputs files, we should delete the H and M-related input parameters to not mislead (even if they are commented out).  This includes the fields_to_plot line below.  And the mag_alpha/gamma stuff too.

File: Docs/source/running_cpp/parameters.rst [hunk 2]
```diff
@@ -878,6 +878,8 @@ Laser initialization
     input file. For a two-dimensional simulation, it is assumed that the first dimension     is `x` and the second dimension in `z`, and the value of `y` is set to zero.
     Note that the current implementation of the parser for external B-field
     does not work with RZ and the code will abort with an error message.
+    Note that the implementation of the parser for external B-field does not work
+    with LLG and the code will abort with an error message
```
Comment (ajnonaka) [inline Docs/source/running_cpp/parameters.rst]:
```suggestion
    with USE_LLG=TRUE and the code will abort with an error message
```

File: Docs/source/running_cpp/parameters.rst [hunk 3]
```diff
@@ -904,13 +906,47 @@ Laser initialization
...
+
 * ``warpx.E_external_grid`` & ``warpx.B_external_grid`` (list of `3 floats`)
     required when ``warpx.E_ext_grid_init_style="constant"``
     and when ``warpx.B_ext_grid_init_style="constant"``, respectively.
     External uniform and constant electrostatic and magnetostatic field added
     to the grid at initialization. Use with caution as these fields are used for
     the field solver. In particular, do not use any other boundary condition
     than periodic.
+    Note that the implementation of the parser for external B-field does not work
+    with LLG and the code will abort with an error message
```
Comment (ajnonaka) [inline Docs/source/running_cpp/parameters.rst]:
```suggestion
    with USE_LLG=TRUE and the code will abort with an error message
```

File: Examples/Tests/Macroscopic_Maxwell/inputs_3d_PR10 (hunk 1 above)
Comment (jackieyao0114) [reply Examples/Tests/Macroscopic_Maxwell/inputs_3d_PR10:72]: agreed. The H, M, alpha and Ms part has been deleted. Thank you!

**Review by ajnonaka — APPROVED** [review]

File: Examples/Tests/Macroscopic_Maxwell/inputs_3d_LLG_PR10 [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+warpx.Ex_external_grid_function(x,y,z) = 0.
+warpx.Ey_external_grid_function(x,y,z) = 0.
+warpx.Ez_external_grid_function(x,y,z) = 0.
+
+warpx.E_excitation_on_grid_style = "parse_E_excitation_grid_function"
+warpx.Ex_excitation_grid_function(x,y,z,t) = "0.0"
+warpx.Ey_excitation_grid_function(x,y,z,t) = "1000*(exp(-(t-3*TP)**2/(2*TP**2))*cos(2*pi*c/wavelength*t)) * (z>=h) * (z<h+6.0e-4)"
+warpx.Ez_excitation_grid_function(x,y,z,t) = "0.0"
+warpx.Ex_excitation_flag_function(x,y,z) = "flag_none"
+warpx.Ey_excitation_flag_function(x,y,z) = "flag_hs * (z>=h) * (z<h+6.0e-4) + flag_none * (z<h) * (z>=h+6.0e-4)"
```
Comment (RevathiJambunathan) [inline Examples/Tests/Macroscopic_Maxwell/inputs_3d_LLG_PR10]:
```suggestion
warpx.Ey_excitation_flag_function(x,y,z) = "flag_hs * (z>=h) * (z<h+6.0e-4) + flag_none * (z<h)  + flag_none * (z>=h+6.0e-4)"
```

Comment (RevathiJambunathan) [reply Examples/Tests/Macroscopic_Maxwell/inputs_3d_LLG_PR10]: (z<h)*z>(h+6e-4) is always going to be false.

File: Examples/Tests/Macroscopic_Maxwell/inputs_3d_LLG_bulkmu [hunk 5]
```diff
@@ -69,6 +75,9 @@ warpx.E_excitation_on_grid_style = "parse_E_excitation_grid_function"
 warpx.Ex_excitation_grid_function(x,y,z,t) = "0.0"
 warpx.Ey_excitation_grid_function(x,y,z,t) = "1000*(exp(-(t-3*TP)**2/(2*TP**2))*cos(2*pi*c/wavelength*t)) * (z>=h) * (z<h+6.0e-4)"
 warpx.Ez_excitation_grid_function(x,y,z,t) = "0.0"
+warpx.Ex_excitation_flag_function(x,y,z) = "flag_none"
+warpx.Ey_excitation_flag_function(x,y,z) = "flag_hs * (z>=h) * (z<h+6.0e-4) + flag_none * (z<h) * (z>=h+6.0e-4)"
```
Comment (RevathiJambunathan) [inline Examples/Tests/Macroscopic_Maxwell/inputs_3d_LLG_bulkmu]:
```suggestion
warpx.Ey_excitation_flag_function(x,y,z) = "flag_hs * (z>=h) * (z<h+6.0e-4) + flag_none * (z<h)  + flag_none * (z>=h+6.0e-4)"
```

File: Examples/Tests/Macroscopic_Maxwell/inputs_3d_LLG_falseB [hunk 6]
```diff
@@ -1,3 +1,7 @@
+# This is a E+B (use USE_LLG=TRUE) simulation of a
+# plane wave in a periodic domain
+# !! Wrong Initialization: B_ext instead of H; thus this input is just for error display
```
Comment (RevathiJambunathan) [inline Examples/Tests/Macroscopic_Maxwell/inputs_3d_LLG_falseB]: wondering if the file should be renamed to `inputs_3d_LLG_abortB`

File: Examples/Tests/Macroscopic_Maxwell/inputs_3d_LLG_original [hunk 7]
```diff
@@ -0,0 +1,134 @@
+####################################################################################################
+## This input file include an uncoupled system to test the M evolution driven by an H_bias field. ##
```
Comment (RevathiJambunathan) [inline Examples/Tests/Macroscopic_Maxwell/inputs_3d_LLG_original]:
```suggestion
## This input file includes an uncoupled system to test the M evolution driven by an H_bias field. ##
```

File: Examples/Tests/Macroscopic_Maxwell/inputs_3d_LLG_original [hunk 8]
```diff
@@ -0,0 +1,134 @@
...
+my_constants.pi = 3.14159265359
+my_constants.L = 141.4213562373095e-6
+my_constants.c = 299792458.
+my_constants.wavelength = 1.2e-1
+
+warpx.E_ext_grid_init_style = parse_E_ext_grid_function
+
+warpx.Ex_external_grid_function(x,y,z) = 0.
+warpx.Ey_external_grid_function(x,y,z) = 0.
+#warpx.Ey_external_grid_function(x,y,z) = "1.e5*exp(-z**2/L**2)*cos(2*pi*z/wavelength)"
```
Comment (RevathiJambunathan) [inline Examples/Tests/Macroscopic_Maxwell/inputs_3d_LLG_original]:
```suggestion
```

File: Examples/Tests/Macroscopic_Maxwell/inputs_3d_LLG_original [hunk 9]
```diff
@@ -0,0 +1,134 @@
...
+warpx.Ez_external_grid_function(x,y,z) = 0.
+
+warpx.H_ext_grid_init_style = parse_H_ext_grid_function
+warpx.Hx_external_grid_function(x,y,z)= 0.
+#warpx.Hx_external_grid_function(x,y,z)= 0.
+#warpx.Hx_external_grid_function(x,y,z)= "-1.e5*exp(-z**2/L**2)*cos(2*pi*z/wavelength)/c/1.25663706212e-06"
```
Comment (RevathiJambunathan) [inline Examples/Tests/Macroscopic_Maxwell/inputs_3d_LLG_original]:
```suggestion
```

File: Examples/Tests/Macroscopic_Maxwell/inputs_3d_LLG_original [hunk 10]
```diff
@@ -0,0 +1,134 @@
...
+## mag_M_normalization==0                    unsaturated materials, i.e. M_magnitude <= mag_Ms.   ##
+## mag_M_normalization==1                    saturated materials, i.e. M_magnitude = mag_Ms.      ##
+##                                           in the second-order algorithm, normalize M_magnitude ##
+##                                           after each iteration in the middle of one time step. ##
+## mag_M_normalization==2                    saturated materials, i.e. M_magnitude = mag_Ms.      ##
+##                                           in the second-order algorithm, normalize M_magnitude ##
+##                                           after all the iterations converge at the end of each ##
+##                                           time step.                                           ##
+## mag_M_normalization==1 is equivalent to mag_M_normalization==2 in the first-order algorithm.   ##
+## This input file requires USE_LLG=TRUE in the GNUMakefile.                                      ##
```
Comment (RevathiJambunathan) [inline Examples/Tests/Macroscopic_Maxwell/inputs_3d_LLG_original:21]:
```suggestion
## This input file requires USE_LLG=TRUE in the GNUMakefile.                                      ##
## Note that the E and H fields are set to zero even though the system is uncoupled to   ##
## prevent the code from running into NANs for this uncoupled case.                              ##
```

File: Examples/Tests/Macroscopic_Maxwell/inputs_3d_LLG_original [hunk 11]
```diff
@@ -0,0 +1,134 @@
...
+warpx.Hz_external_grid_function(x,y,z) = 0.
+
+#unit conversion: 1 Gauss = 1 Oersted = (1000/4pi) A/m
+#calculation of H_bias: H_bias (oe) = frequency / 2.8e6
+
+warpx.H_bias_ext_grid_init_style = parse_H_bias_ext_grid_function
+warpx.Hx_bias_external_grid_function(x,y,z)= 0.
+warpx.Hy_bias_external_grid_function(x,y,z)= 0.
+warpx.Hz_bias_external_grid_function(x,y,z)= 3e4 # in A/m, equal to 382 Oersted
+#warpx.Hz_bias_external_grid_function(x,y,z)= 0. # in A/m, equal to 382 Oersted
```
Comment (RevathiJambunathan) [inline Examples/Tests/Macroscopic_Maxwell/inputs_3d_LLG_original]:
```suggestion

```

File: Examples/Tests/Macroscopic_Maxwell/inputs_3d_LLG_original [hunk 12]
```diff
@@ -0,0 +1,134 @@
...
+
+#warpx.H_bias_ext_grid_init_style = constant
+#warpx.Hx_bias_external_grid = 0
+#warpx.Hy_bias_external_grid = 0
+#warpx.Hz_bias_external_grid = 0
+
+#warpx.M_ext_grid_init_style = parse_M_ext_grid_function
+#warpx.Mx_external_grid_function(x,y,z)= "-1.e5*exp(-z**2/L**2)*cos(2*pi*z/wavelength)/c"
+#warpx.My_external_grid_function(x,y,z)= 0.
+#warpx.Mz_external_grid_function(x,y,z) = 0.
```
Comment (RevathiJambunathan) [inline Examples/Tests/Macroscopic_Maxwell/inputs_3d_LLG_original]:
```suggestion
```

File: Examples/Tests/Macroscopic_Maxwell/inputs_3d_LLG_original [hunk 13]
```diff
@@ -0,0 +1,134 @@
...
+macroscopic.mag_Ms_function(x,y,z) = "1.4e5"
+#macroscopic.mag_Ms = 1.4e5 # in unit A/m, equal to 1750 Gauss
+
+macroscopic.mag_alpha_init_style = "parse_mag_alpha_function" # parse or "constant"
+macroscopic.mag_alpha_function(x,y,z) = "5e-01"
+#macroscopic.mag_alpha = 5e-01 # alpha is unitless, typical values range from 1e-3 ~ 1e-5
+
+macroscopic.mag_gamma_init_style = "parse_mag_gamma_function" # parse or "constant"
+macroscopic.mag_gamma_function(x,y,z) = "-1.759e11"
+#macroscopic.mag_gamma = -1.759e11 # gyromagnetic ratio is constant for electrons in all materials
```
Comment (RevathiJambunathan) [inline Examples/Tests/Macroscopic_Maxwell/inputs_3d_LLG_original]:
```suggestion
```

File: Examples/Tests/Macroscopic_Maxwell/inputs_3d_LLG_original [hunk 14]
```diff
@@ -0,0 +1,134 @@
...
+#macroscopic.mu = 1.25663706212e-06
+
+#unit conversion: 1 Gauss = (1000/4pi) A/m
+macroscopic.mag_Ms_init_style = "parse_mag_Ms_function" # parse or "constant"
+macroscopic.mag_Ms_function(x,y,z) = "1.4e5"
+#macroscopic.mag_Ms = 1.4e5 # in unit A/m, equal to 1750 Gauss
+
+macroscopic.mag_alpha_init_style = "parse_mag_alpha_function" # parse or "constant"
+macroscopic.mag_alpha_function(x,y,z) = "5e-01"
+#macroscopic.mag_alpha = 5e-01 # alpha is unitless, typical values range from 1e-3 ~ 1e-5
```
Comment (RevathiJambunathan) [inline Examples/Tests/Macroscopic_Maxwell/inputs_3d_LLG_original]:
```suggestion
```

File: Examples/Tests/Macroscopic_Maxwell/inputs_3d_LLG_original [hunk 15]
```diff
@@ -0,0 +1,134 @@
...
+#macroscopic.epsilon = 8.8541878128e-12
+
+macroscopic.mu_init_style = "parse_mu_function" # parse or "constant"
+macroscopic.mu_function(x,y,z) = "1.25663706212e-06"
+#macroscopic.mu = 1.25663706212e-06
+
+#unit conversion: 1 Gauss = (1000/4pi) A/m
+macroscopic.mag_Ms_init_style = "parse_mag_Ms_function" # parse or "constant"
+macroscopic.mag_Ms_function(x,y,z) = "1.4e5"
+#macroscopic.mag_Ms = 1.4e5 # in unit A/m, equal to 1750 Gauss
```
Comment (RevathiJambunathan) [inline Examples/Tests/Macroscopic_Maxwell/inputs_3d_LLG_original]:
```suggestion
```

File: Examples/Tests/Macroscopic_Maxwell/inputs_3d_LLG_original [hunk 16]
```diff
@@ -0,0 +1,134 @@
...
+algo.em_solver_medium = macroscopic # vacuum/macroscopic
+
+algo.macroscopic_sigma_method = laxwendroff # laxwendroff or backwardeuler
+macroscopic.sigma_init_style = "parse_sigma_function" # parse or "constant"
+macroscopic.sigma_function(x,y,z) = "0.0"
+#macroscopic.sigma = 0.0
+
+macroscopic.epsilon_init_style = "parse_epsilon_function" # parse or "constant"
+macroscopic.epsilon_function(x,y,z) = "8.8541878128e-12"
+#macroscopic.epsilon = 8.8541878128e-12
```
Comment (RevathiJambunathan) [inline Examples/Tests/Macroscopic_Maxwell/inputs_3d_LLG_original]:
```suggestion
```

File: Examples/Tests/Macroscopic_Maxwell/inputs_3d_LLG_original [hunk 17]
```diff
@@ -0,0 +1,134 @@
...
+warpx.mag_M_normalization = 2
+warpx.mag_LLG_coupling = 0
+particles.nspecies = 0
+
+algo.em_solver_medium = macroscopic # vacuum/macroscopic
+
+algo.macroscopic_sigma_method = laxwendroff # laxwendroff or backwardeuler
+macroscopic.sigma_init_style = "parse_sigma_function" # parse or "constant"
+macroscopic.sigma_function(x,y,z) = "0.0"
+#macroscopic.sigma = 0.0
```
Comment (RevathiJambunathan) [inline Examples/Tests/Macroscopic_Maxwell/inputs_3d_LLG_original]:
```suggestion
```

File: Examples/Tests/Macroscopic_Maxwell/inputs_3d_LLG_original [hunk 18]
```diff
@@ -0,0 +1,134 @@
...
+
+macroscopic.epsilon_init_style = "parse_epsilon_function" # parse or "constant"
+macroscopic.epsilon_function(x,y,z) = "8.8541878128e-12"
+#macroscopic.epsilon = 8.8541878128e-12
+
+macroscopic.mu_init_style = "parse_mu_function" # parse or "constant"
+macroscopic.mu_function(x,y,z) = "1.25663706212e-06"
+#macroscopic.mu = 1.25663706212e-06
```
Comment (RevathiJambunathan) [inline Examples/Tests/Macroscopic_Maxwell/inputs_3d_LLG_original]:
```suggestion
```

File: Examples/Tests/Macroscopic_Maxwell/inputs_3d_LLG_periodic_noMs [hunk 19]
```diff
@@ -0,0 +1,68 @@
+# This is a E+H (use USE_LLG=TRUE but Ms=0, i.e. no LLG) simulation of a
```
Comment (RevathiJambunathan) [inline Examples/Tests/Macroscopic_Maxwell/inputs_3d_LLG_periodic_noMs]:
```suggestion
# This is a E+H (use USE_LLG=TRUE but Ms=0, i.e. no evolution of M using the LLG module) simulation of a
```

Comment (RevathiJambunathan) [reply Examples/Tests/Macroscopic_Maxwell/inputs_3d_LLG_periodic_noMs]: we still use the LLG module, but within the module a determination is made to update M or not?

File: Examples/Tests/Macroscopic_Maxwell/inputs_3d_LLG_tfmu [hunk 20]
```diff
@@ -75,8 +81,11 @@ warpx.Ez_external_grid_function(x,y,z) = 0.
 
 warpx.E_excitation_on_grid_style = "parse_E_excitation_grid_function"
 warpx.Ex_excitation_grid_function(x,y,z,t) = "0.0"
-warpx.Ey_excitation_grid_function(x,y,z,t) = "(exp((t-3*TP)**2/(2*TP**2))*cos(2*pi*c/wavelength*t)) * (z>=h) * (z<h+1.0e-4)"
+warpx.Ey_excitation_grid_function(x,y,z,t) = "(exp((t-3*TP)**2/(2*TP**2))*cos(2*pi*c/wavelength*t)) * (z>=h) * (z<h+6.0e-4)"
 warpx.Ez_excitation_grid_function(x,y,z,t) = "0.0"
+warpx.Ex_excitation_flag_function(x,y,z) = "flag_none"
+warpx.Ey_excitation_flag_function(x,y,z) = "flag_hs * (z>=h) * (z<h+6.0e-4) + flag_none * (z<h) * (z>=h+6.0e-4)"
```
Comment (RevathiJambunathan) [inline Examples/Tests/Macroscopic_Maxwell/inputs_3d_LLG_tfmu]:
```suggestion
warpx.Ey_excitation_flag_function(x,y,z) = "flag_hs * (z>=h) * (z<h+6.0e-4) + flag_none * (z<h)  + flag_none* (z>=h+6.0e-4)"
```

File: Examples/Tests/Macroscopic_Maxwell/inputs_3d_LLG_uncoupled [hunk 21]
```diff
@@ -72,13 +72,6 @@ warpx.Ey_external_grid_function(x,y,z) = 0.
 #warpx.Ey_external_grid_function(x,y,z) = "1.e5*exp(-z**2/L**2)*cos(2*pi*z/wavelength)"
```
Comment (RevathiJambunathan) [inline Examples/Tests/Macroscopic_Maxwell/inputs_3d_LLG_uncoupled]:
```suggestion
```

File: Examples/Tests/Macroscopic_Maxwell/inputs_3d_LLG_uncoupled_unsat [hunk 22]
```diff
@@ -72,13 +72,6 @@ warpx.Ey_external_grid_function(x,y,z) = 0.
 #warpx.Ey_external_grid_function(x,y,z) = "1.e5*exp(-z**2/L**2)*cos(2*pi*z/wavelength)"
```
Comment (RevathiJambunathan) [inline Examples/Tests/Macroscopic_Maxwell/inputs_3d_LLG_uncoupled_unsat]:
```suggestion
```

File: Examples/Tests/Macroscopic_Maxwell/inputs_3d_PR10 [hunk 23]
```diff
@@ -0,0 +1,81 @@
...
+warpx.Ex_external_grid_function(x,y,z) = 0.
+warpx.Ey_external_grid_function(x,y,z) = 0.
+warpx.Ez_external_grid_function(x,y,z) = 0.
+
+warpx.E_excitation_on_grid_style = "parse_E_excitation_grid_function"
+warpx.Ex_excitation_grid_function(x,y,z,t) = "0.0"
+warpx.Ey_excitation_grid_function(x,y,z,t) = "1000*(exp(-(t-3*TP)**2/(2*TP**2))*cos(2*pi*c/wavelength*t)) * (z>=h) * (z<h+6.0e-4)"
+warpx.Ez_excitation_grid_function(x,y,z,t) = "0.0"
+warpx.Ex_excitation_flag_function(x,y,z) = "flag_none"
+warpx.Ey_excitation_flag_function(x,y,z) = "flag_hs * (z>=h) * (z<h+6.0e-4) + flag_none * (z<h) * (z>=h+6.0e-4)"
```
Comment (RevathiJambunathan) [inline Examples/Tests/Macroscopic_Maxwell/inputs_3d_PR10]:
```suggestion
warpx.Ey_excitation_flag_function(x,y,z) = "flag_hs * (z>=h) * (z<h+6.0e-4) + flag_none * (z<h) + flag_none *(z>=h+6.0e-4)"
```

**Review by RevathiJambunathan — COMMENTED** [review]
Thank you @jackieyao0114 for this PR. 
I mostly have clean-up suggestions in the places where I saw commented out lines. There were other places that had these commented out lines, but I could not change them through the github interface since the PR did not "touch/modify" those lines.

The only other comment is the way "flag none" is used in the input for the excitation.

File: Examples/Tests/Macroscopic_Maxwell/inputs_3d_LLG_falseB (hunk 6 above)
Comment (jackieyao0114) [reply Examples/Tests/Macroscopic_Maxwell/inputs_3d_LLG_falseB]: sure!

File: Examples/Tests/Macroscopic_Maxwell/inputs_3d_LLG_periodic_noMs (hunk 19 above)
Comment (jackieyao0114) [reply Examples/Tests/Macroscopic_Maxwell/inputs_3d_LLG_periodic_noMs]: yes~

File: Examples/Tests/Macroscopic_Maxwell/inputs_3d_LLG_PR10 (hunk 4 above)
Comment (jackieyao0114) [reply Examples/Tests/Macroscopic_Maxwell/inputs_3d_LLG_PR10]: great point! Thank you so much!

**Review by RevathiJambunathan — APPROVED** [review]

**Pull request merged**